### PR TITLE
Set Debian's default locale to be C.UTF-8

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
Set Debian's default locale to be C.UTF-8

When no locale is found in /etc/default/locale, cloud-init was
defaulting to generating en_US.UTF-8. This can be an expensive operation
and isn't necessary. Instead, change the default to C.UTF-8 which
generally shouldn't require generation.

Also add some logging to make it more obvious where cloud-init is
getting the locale information from.

LP: #2038945
```

## Checklist:

 - [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/distros/debian.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def __init__(self, name, cfg, paths):
         # calls from repeatly happening (when they
         # should only happen say once per instance...)
         self.osfamily = "debian"
-        self.default_locale = "en_US.UTF-8"
+        self.default_locale = "c.UTF-8"
```
Comment (mwhudson) [inline cloudinit/distros/debian.py:57]: I'm fairly sure it doesn't matter but can you spell this "C.UTF-8" please!

File: cloudinit/distros/debian.py [hunk 2]
```diff
@@ -66,10 +66,21 @@ def get_locale(self):
...
-        )
+        if self.system_locale:
+            LOG.debug(
+                "System locale set to %s via %s",
+                self.system_locale,
+                LOCALE_CONF_FN,
+            )
+            return self.system_locale
+        else:
+            LOG.debug(
```
Comment (blackboxsw) [inline cloudinit/distros/debian.py:77]: Changeset looks good, but we have no unittest coverage for this path of the conditional

**Review by blackboxsw — APPROVED** [review]
Changset looks good to set locale defaults. Let's also get review from Noah on this as debian could prefer images to retain a default of `en_US.UTF-8`  if the image itself didn't provide a default in /etc/default/locale (or by proxy /etc/locale.conf).

File: cloudinit/distros/debian.py (hunk 2 above)
Comment (blackboxsw) [reply cloudinit/distros/debian.py:77]: tested on images w/out /etc/default/locale. Success regenerating locales in absence of system defaults

**Comment by TheRealFalcon** [comment]
@nmeyerhans Would you mind reviewing this?

File: cloudinit/distros/debian.py (hunk 2 above)
Comment (TheRealFalcon) [reply cloudinit/distros/debian.py:77]: Looking at this more, I may move the logging around. In the case where no locale has been specified, `get_locale` is going to get called twice so we'll see duplicate logs

**Comment by TheRealFalcon** [comment]
@blackboxsw I moved the log messages so that they don't log multiple times. I also added unit tests. One commit is converting the tests to pytest. The other commit is the actual test changes.

**Review by blackboxsw — APPROVED** [review]
Approved w/ unittest coverage and logging messages are only written once

**Review by nmeyerhans — APPROVED** [review]
👍 to changing the default locale; it is consistent with what we set in the Debian cloud images, so should be a no-op in most cases.

**Comment by holmanb** [comment]
> Also, tests require a fixed version of #4499 . Without it we get no logs from caplog.

+1 Merged

**Pull request merged**